### PR TITLE
[CH-218]Clickhouse supports window function

### PR DESCRIPTION
**Pull request by lgbo-ustc**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

Clickhouse supports window function now.

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
unit test

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)
test by CH[[235]]

### CODE REVIEW

File: backends-clickhouse/src/test/scala/io/glutenproject/execution/GlutenClickHouseTPCHSuite.scala [hunk 1]
```diff
@@ -327,4 +327,5 @@ class GlutenClickHouseTPCHSuite extends GlutenClickHouseTPCHAbstractSuite {
       ("spark.sql.autoBroadcastJoinThreshold", "-1"),
       ("spark.gluten.sql.columnar.forceshuffledhashjoin", "true")) {}
   }
+
```
Comment (zzcclp) [inline backends-clickhouse/src/test/scala/io/glutenproject/execution/GlutenClickHouseTPCHSuite.scala:330]: revert

File: gluten-core/src/main/java/io/glutenproject/substrait/expression/WindowFunctionNode.java [hunk 2]
```diff
@@ -54,7 +54,7 @@ public class WindowFunctionNode implements Serializable {
 
   private Expression.WindowFunction.Bound.Builder setBound(
       Expression.WindowFunction.Bound.Builder builder,
-      String type) {
+      String type, boolean isLowerBound) {
```
[the remark this answers is not in the document]
Comment (lgbo-ustc) [reply gluten-core/src/main/java/io/glutenproject/substrait/expression/WindowFunctionNode.java:57]: this is not frame type. type of bound

**Comment by lgbo-ustc** [comment]
@PHILO-HE please check what is wrong with velox ?

**Comment by lgbo-ustc** [comment]
This seems to be a random bug
```
22/12/26 02:03:40 ERROR Executor: Exception in task 0.0 in stage 68.0 (TID 73)
java.lang.RuntimeException: unordered_map::at: key not found
0. std::logic_error::logic_error(char const*) @ 0x16a3ae94 in /tmp/libch.so
1. std::out_of_range::out_of_range(char const*) @ 0xa7fa769 in /tmp/libch.so
2. std::__1::__throw_out_of_range(char const*) @ 0xa7fa71f in /tmp/libch.so
3. local_engine::StorageMergeTreeFactory::getStorage(DB::StorageID, DB::ColumnsDescription, std::__1::function<std::__1::shared_ptr<local_engine::CustomStorageMergeTree> ()>) @ 0xa891c0a in /tmp/libch.so
4. local_engine::SerializedPlanParser::parseMergeTreeTable(substrait::ReadRel const&) @ 0xa81486d in /tmp/libch.so
5. local_engine::SerializedPlanParser::parseOp(substrait::Rel const&) @ 0xa81bcdb in /tmp/libch.so
6. local_engine::SerializedPlanParser::parse(std::__1::unique_ptr<substrait::Plan, std::__1::default_delete<substrait::Plan> >) @ 0xa81aa22 in /tmp/libch.so
7. local_engine::SerializedPlanParser::parse(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) @ 0xa824fb8 in /tmp/libch.so
8. Java_io_glutenproject_vectorized_ExpressionEvaluatorJniWrapper_nativeCreateKernelWithIterator @ 0xa8d3486 in /tmp/libch.so

	at io.glutenproject.vectorized.ExpressionEvaluatorJniWrapper.nativeCreateKernelWithIterator(Native Method)
	at io.glutenproject.vectorized.NativeExpressionEvaluator.createKernelWithBatchIterator(NativeExpressionEvaluator.java:57)
	at io.glutenproject.backendsapi.clickhouse.CHIteratorApi.genFirstStageIterator(CHIteratorApi.scala:171)
	at io.glutenproject.execution.GlutenWholeStageColumnarRDD.compute(GlutenWholeStageColumnarRDD.scala:113)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:131)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:506)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1491)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:509)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
```

**Comment by zzcclp** [comment]
re-run again.

**Comment by PHILO-HE** [comment]
> @PHILO-HE please check what is wrong with velox ?

Currently, velox doesn't support expression/literal bound type. There is a bug in velox validator. I am fixing it. Thanks!

**Comment by zzcclp** [comment]
@lgbo-ustc @PHILO-HE can we ingore the failed velox test cases first, after fixing these test cases, open them?

**Comment by lgbo-ustc** [comment]
@PHILO-HE shoule we disable window support for velox at first or wait for your fix?

**Comment by PHILO-HE** [comment]
I have made some fix in velox. Just triggered the velox job again. Let's wait for the feedback. Thanks!

**Review by PHILO-HE — APPROVED** [review]
The PR looks good. Thanks!

**Review by zzcclp — APPROVED** [review]
LGTM

**Pull request merged**